pkg/controllers/user/dnsrecord/dnsrecord.go

```diff
@@ -172,3 +221,30 @@ func (c *EndpointController) reconcileServicesForEndpoint(key string, obj *corev
 
 	return nil
 }
+
+func (c *Controller) queueUpdateForExtNameSvc(key string, delete bool) {
+	alias, ok := serviceUUIDToHostNameAlias.Load(key)
+	if ok && delete {
+		serviceUUIDToHostNameAlias.Delete(key)
+	}
+	if ok {
```

Comment:
we do this section even if delete is true. Any concerns with the race condition, when delete from the map happens, but enqueue fails?